hawkular-javaagent/src/main/java/org/hawkular/agent/javaagent/cmd/UpdateCollectionIntervalsCommand.java

```diff
@@ -0,0 +1,199 @@
+                String[] names = parseAvailTypeId(availTypeId);
+                String availSetName = names[0];
+                String availName = names[1];
+                // find the avail and change its interval if found
+                for (DMRAvailSet availSet : javaAgentConfig.getDmrAvailSets()) {
+                    if (availSetName.equals(availSet.getName())) {
+                        for (DMRAvail avail : availSet.getDmrAvails()) {
+                            if (availName.equals(avail.getName())) {
+                                avail.setInterval(Integer.valueOf(entry.getValue()));
+                                avail.setTimeUnits(TimeUnits.seconds); // the command always assumes seconds
+                                requireRestart = true;
+                                continue NEXT_AVAIL;
+                            }
+                        }
+                    }
+                }
+                for (JMXAvailSet availSet : javaAgentConfig.getJmxAvailSets()) {
+                    if (availSetName.equals(availSet.getName())) {
+                        for (JMXAvail avail : availSet.getJmxAvails()) {
+                            if (availName.equals(avail.getName())) {
+                                avail.setInterval(Integer.valueOf(entry.getValue()));
+                                avail.setTimeUnits(TimeUnits.seconds); // the command always assumes seconds
+                                requireRestart = true;
+                                continue NEXT_AVAIL;
+                            }
+                        }
+                    }
+                }
+            }
+        }
+
+        if (requireRestart) {
+            context.addResponseSentListener(new ResponseSentListener() {
+                @Override
+                public void onSend(BasicMessageWithExtraData<? extends BasicMessage> response, Exception sendError) {
+                    log.info("Collection intervals updated. Persisting changes and restarting agent.");
+                    javaAgent.stopHawkularAgent();
+                    javaAgent.startHawkularAgent(javaAgentConfig);
+                }
+            });
+        } else {
+            log.debug("Skipping collection interval update, no valid type updates provided.");
+        }
+
+        return null;
+    }
+
+    @Override
+    protected void validate(BasicMessageWithExtraData<UpdateCollectionIntervalsRequest> envelope,
+            MonitoredEndpoint<? extends AbstractEndpointConfiguration> endpoint) {
+        return; // no-op
+    }
+
+    private String[] parseMetricTypeId(String metricTypeId) {
+        String[] names = metricTypeId.split("~");
+        if (names.length != 2) {
+            throw new IllegalArgumentException(
+                    "MetricTypeId must be of form MetricTypeSetName~MetricTypeName: " + metricTypeId);
+        }
+        return names;
+    }
+
+    private String[] parseAvailTypeId(String availTypeId) {
```

Comment:
So it's same as parseMetricTypeId, is it kept separate on purpose? (maybe more validation will be added later?)

Reply:
Yes, perhaps this should be revisited it, I just brought back this class but definitely not all tests are passing yet.